cmd/copyUploadEnumerator.go

```diff
@@ -64,6 +69,9 @@ func (e *copyUploadEnumerator) enumerate(cca *cookedCopyCmdArgs) error {
 			}
 
 			cleanedSource := strings.Replace(listOfFilesAndDirectories[0], common.OS_PATH_SEPARATOR, common.AZCOPY_PATH_SEPARATOR_STRING, -1)
+			if destinationURL.Path[len(destinationURL.Path)-1:] == `/` {
```

Comment:
hmmm I'm a bit confused, why isn't line 53-55 covering this case already?

Reply:
urlIsContainerOrShare is... a really odd function to start with. It doesn't even do the job that the rules listed on top of the function even describe in the first place. I'm rewriting it.